src/controllers/activegate/reconciler/statefulset/reconciler.go

```diff
@@ -236,3 +265,22 @@ func (r *Reconciler) getDataFromCustomProperty(customProperties *dynatracev1beta
 	}
 	return customProperties.Value, nil
 }
+
+func (r *Reconciler) getDataFromAuthTokenSecret() (string, error) {
```

Comment:
Nit: make it generic, move to `kubeobjects`, like:
`getDataFromSecret(secretName string, namespace string) (string, error)`